src/core/comparator.ml

```diff
@@ -3,26 +3,26 @@
 type 'a t = 'a -> 'a -> Ordering.t
 
 let make (compare: 'a -> 'a -> int) : 'a t = fun this that ->
-    let r = compare this that in 
-    if r = 0 then Ordering.Equal
-    else begin
-        if r > 0 then Ordering.Greater
-        else Ordering.Less
-    end 
+  let r = compare this that in 
+  if r = 0 then Ordering.Equal
+  else begin
+    if r > 0 then Ordering.Greater
+    else Ordering.Less
+  end 
 
 let bytes = make Bytes.compare
 let char = make Char.compare
 let int (this : int) (that : int) =
-    (if this < that
-     then Ordering.Less
-     else if this > that then Ordering.Greater else Ordering.Equal : 
-                                                        Ordering.t)
+  (if this < that
+   then Ordering.Less
+   else if this > that then Ordering.Greater else Ordering.Equal : 
+                            Ordering.t)
 let int32 = make Int32.compare
 let int64 = make Int64.compare
 let nativeInt = make Nativeint.compare
 let string = make String.compare
 
 let toEquality (comparator : 'a t) =
-    (fun x  ->
-         fun y  -> if (comparator x y) == Ordering.equal then true else false : 
-                                                                            'a Equality.t)
\ No newline at end of file
+  (fun x  ->
+     fun y  -> if (comparator x y) == Ordering.equal then true else false : 
+                                      'a Equality.t)
```

Comment:
I don't understand why this is explicitly curried, it would probably format better if it was not. Also weird spacing before the arrows.